File: server/internal/model/stats.go
/*
项目名称：JeriBlog
文件名称：stats.go
创建时间：2026-04-16 15:00:36

系统用户：Jerion
作　　者：Jerion
联系邮箱：416685476@qq.com
功能描述：统计数据模型
*/

package model

import "time"

// Visit 访问记录模型
type Visit struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	VisitorID string    `gorm:"size:64;not null" json:"visitor_id"`         // 访客唯一标识
	IP        string    `gorm:"size:45;not null" json:"ip"`                 // 访客IP
	PageURL   string    `gorm:"size:500" json:"page_url"`                   // 访问页面URL
	ArticleID *uint     `json:"article_id"`                                 // 文章ID（可选）
	UserAgent string    `gorm:"type:text" json:"user_agent"`                // 浏览器UA
	Location  string    `gorm:"size:100" json:"location"`                   // 地理位置
	Browser   string    `gorm:"size:50" json:"browser"`                     // 浏览器
	OS        string    `gorm:"size:50" json:"os"`                          // 操作系统
	Referer   string    `gorm:"size:500" json:"referer"`                    // 来源页面
	VisitDate time.Time `gorm:"type:date;not null;index" json:"visit_date"` // 访问日期
	CreatedAt time.Time `json:"created_at"`
}

File: admin/src/types/stats.ts
/*
项目名称：JeriBlog
文件名称：stats.ts
创建时间：2026-04-16 15:08:10

系统用户：Jerion
作　　者：Jerion
联系邮箱：416685476@qq.com
功能描述：类型定义 - stats类型
*/

// Dashboard 统计数据
export interface DashboardStats {
  total_articles: number
  total_friends: number
  total_moments: number
  total_views: number
  total_visitors: number
  total_comments: number
  total_users: number
  today_views: number
  today_visitors: number
  today_comments: number
  today_users: number
  views_growth: number
  visitors_growth: number
  comments_growth: number
  users_growth: number
}

// 趋势数据项
export interface TrendDataItem {
  date: string
  pv_count: number
  uv_count: number
}

// 趋势查询参数
export interface TrendQuery {
  start_date: string
  end_date: string
  type: 'daily' | 'monthly'
}

// 标签统计数据
export interface TagStats {
  name: string
  count: number
}

// 文章贡献数据
export interface ArticleContribution {
  date: string
  count: number
}

// 文章贡献查询参数
export interface ContributionQuery {
  year?: number  // 年份（可选）
  month?: number // 月份 1-12（可选）
}

// 分类统计数据
export interface CategoryStats {
  name: string
  count: number
}

// 访问日志实体
export interface Visit {
  id: number
  visitor_id: string
  ip: string
  page_url: string
  user_agent: string
  location: string
  browser: string
  os: string
  referer: string
  created_at: string
}

// 访问日志列表数据
export interface VisitListData {
  list: Visit[]
  total: number
  page: number
  page_size: number
}

// 访问日志查询参数
export interface VisitQuery {
  page: number
  page_size: number
  keyword?: string
  start_date?: string
  end_date?: string
}
